packages/cli/src/utils/network-check.ts

```diff
@@ -0,0 +1,32 @@
+import { DEFAULT_FETCH_TIMEOUT } from './fetch-with-timeout.js';
+
+/**
+ * Checks if the system has internet connectivity by attempting to reach a reliable endpoint
+ * @param timeout - Timeout in milliseconds for the connectivity check
+ * @returns Promise<boolean> - true if internet is available, false otherwise
+ */
+export async function hasInternetConnectivity(
+  timeout: number = DEFAULT_FETCH_TIMEOUT
+): Promise<boolean> {
+  // Skip network check if explicitly disabled
+  if (process.env.REDOCLY_OFFLINE === 'true' || process.env.CI === 'true') {
+    return false;
+  }
+
+  try {
+    // Use a reliable endpoint for connectivity check
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), timeout);
+
+    const response = await fetch('https://otel.cloud.redocly.com', {
```

Comment:
It'd be great to reuse the const from `otel.ts` (or vice versa -- export this domain and reuse it there).